devito/propagator.py

```diff
@@ -200,7 +201,7 @@ def sympy_to_cgen(self, stencils):
                 # TODO: add support for double precision
                 self.add_local_var(name, np.float32)
                 # TODO: undo precision enforcing
-                factors.append(cgen.Assign(name, str(ccode(expr.xreplace(self._var_map))).replace("pow", "powf").replace("fabs", "fabsf")))
+                factors.append(cgen.Assign(name, str(ccode(self.time_substitutions(expr).xreplace(self._var_map))).replace("pow", "powf").replace("fabs", "fabsf")))
```

Comment:
The old implementation had factors independent of time so it didn't matter. but if your factor is e.g
`alpha = u(t+s, x ,y,z)`  it will not replace the time variable as it is not in `_var_map`